crates/flux-middle/src/fhir.rs

```diff
@@ -725,10 +742,17 @@ impl Ident {
 /// in a definition.
 ///
 /// [early bound]: https://rustc-dev-guide.rust-lang.org/early-late-bound.html
+///
+/// Sort parameters e.g. #[flux::refined_by( elems: Set<T> )] tracks the mapping from
+/// bound Var -> Generic id. e.g. if we have RMap<K, V> refined_by(keys: Set<K>)
+/// then RMapIdx = forall #0. { keys: Set<#0> }
+/// and sort_params = vec![0]  i.e. maps Var(0) to Generic(0)
+
 #[derive(Clone, Debug, TyEncodable, TyDecodable)]
 pub struct RefinedBy {
     pub def_id: DefId,
     pub span: Span,
+    sort_params: Vec<usize>,
```

Comment:
hmm, I don't like this representation too much. I thought this was going to be a `Vec<DefId>`. Related to https://github.com/flux-rs/flux/pull/542#issuecomment-1791049978 perhaps.

In any case, I'm not too worried about this. The code is probably going to have a major refactor anyways.

Reply:
fixed.